bundles/admin/admin-permissions/view/LayerRightsTable.jsx

```diff
@@ -46,66 +39,26 @@ const getPermissionTableHeader = (permission) => {
 };
 
 export const LayerRightsTable = ThemeConsumer(({ theme, controller, state }) => {
-    const searchInput = useRef(null);
-    const getColumnSearchProps = (dataIndex) => ({
-        filterDropdown: ({ setSelectedKeys, selectedKeys, confirm, clearFilters }) => (
-          <FilterContainer onKeyDown={(e) => e.stopPropagation()}>
-            <FilterFields>
-                <TextInput
-                    ref={searchInput}
-                    value={selectedKeys[0]}   
-                    onChange={(e) => setSelectedKeys(e.target.value ? [e.target.value] : [])}
-                    onPressEnter={() => confirm({ closeDropdown: true })}
-                    allowClear
-                />
-            </FilterFields>
-            <Space>
-                <Button
-                    type="primary"
-                    onClick={() => confirm({ closeDropdown: true })}
-                    size="small"
-                >
-                    <Message messageKey='flyout.filter' />
-                </Button>
-                <SecondaryButton
-                    type="clear"
-                    onClick={() => {
-                        clearFilters();
-                        confirm({ closeDropdown: true });
-                    }}
-                    size="small"
-                />
-            </Space>
-          </FilterContainer>
-        ),
-        filterIcon: (filtered) => (
-          <SearchIcon $filtered={filtered} $theme={theme} />
-        ),
-        onFilter: (value, record) =>
-          record[dataIndex]
-            .toString()
-            .toLowerCase()
-            .includes((value).toLowerCase()),
-        onFilterDropdownOpenChange: (visible) => {
-          if (visible) {
-            setTimeout(() => searchInput.current?.select(), 100);
-          }
-        },
-        render: (text) => text
-    });
     const columnSettings = [];
     if (state.permissions?.names) {
         columnSettings.push({
             align: 'left',
             title: <Message messageKey='rights.name' />,
             dataIndex: 'name',
-            sorter: getSorterFor('name'),
-            ...getColumnSearchProps('name')
+            sorter: getSorterFor('name')
         });
         state.permissions.names.forEach((name, index) => {
             columnSettings.push({
                 align: 'left',
-                title: getPermissionTableHeader(name),
+                title: () => (
+                    <HeaderCell>
+                        {getPermissionTableHeader(name)}
+                        <CheckAllCheckbox
+                            checked={state.checkAllCheckboxes[state.pagination.page]?.[name.id]}
+                            onChange={() => controller.setCheckAllForPermission(name.id)}
```

Comment:
What if we didn't store the state of this "check all" but calculated it on the fly using helpers like:
```
const hasPermission = (layer, permissionType) => {
    return layer.permissions.includes(p => p === permissionType)
};
const allChecked = (permissionType) => {
    state.resources.every(layer => hasPermission(layer, permissionType))
};
```
So this could be something like (sending boolean param to setter to see if we should remove the permission or add it):
```
const permissionType = name.id;
const allCurrentLayersHavePermission = allChecked(permissionType);

                        <CheckAllCheckbox
                            checked={allCurrentLayersHavePermission}
                            onChange={() => controller.setPermissionForCurrentLayers(permissionType, !allCurrentLayersHavePermission)}
```

Reply:
Done with small changes. Array.every didn't work here for some reason.